### DISCUSSION TITLE
Shared types in extensions

### DISCUSSION THREAD

**Discussion by Enhed** [directus/directus#16520] [Archive]
Can i get access to shared types? If i use import like this
```ts
import {PrimaryKey, User} from "@directus/shared/types"
```
i got hint `TS2307: Cannot find module '@directus/shared/types' or its corresponding type declarations.` and cant see types. I use bundle extensions and set devDeps with `"@directus/shared": "^9.21.0"`

**Comment by rijkvanzanten** [comment]
Are you building in CJS or ESM? I believe there might be some ESM gotchas in the way the types are exported at the moment
